interceptor.go

```diff
@@ -216,6 +220,11 @@ func (i *Interceptor) WrapStreamingClient(next connect.StreamingClientFunc) conn
 				// streams because the error only exists when finishing the stream.
 				state.addAttributes(statusCodeAttribute(protocol, state.error))
 				span.SetAttributes(state.attributes...)
+				// request metadata attributes are added here because connect interceptors don't have access to request
+				// metadata until the send function is called. This means that traces are created without the request
+				// metadata attributes, so samplers don't have access to metadata attributes in streaming clients.
+				span.SetAttributes(headerAttributes(protocol, requestKey, conn.RequestHeader(), i.config.metadataReqKeys)...)
+				span.SetAttributes(headerAttributes(protocol, responseKey, conn.ResponseHeader(), i.config.metadataResKeys)...)
```

Comment:
Seems like we ought to move span creation to the first call to `Send` or the first call to `Close`, whichever happens first? That'll also make the RPC duration match network activity more closely.

Reply:
Completed through `createSpanOnce` that gets called in both `close` and `send`: 1f41e4cb0dd057c879a3b0e2d839564a194f919e